bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/HeapDumpAnalysisHandler.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright 2021 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.ui.handlers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.expressions.IEvaluationContext;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.ide.IDE;
+import org.eclipse.ui.part.FileEditorInput;
+
+import com.espressif.idf.core.IDFCorePlugin;
+import com.espressif.idf.core.IDFEnvironmentVariables;
+import com.espressif.idf.core.ProcessBuilderFactory;
+import com.espressif.idf.core.logging.Logger;
+import com.espressif.idf.core.util.FileUtil;
+import com.espressif.idf.core.util.IDFUtil;
+import com.espressif.idf.ui.tracing.HeapTracingAnalysisEditor;
+import com.espressif.idf.ui.update.AbstractToolsHandler;
+
+/**
+ * Handler class for handling the context menu action
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class HeapDumpAnalysisHandler extends AbstractToolsHandler
+{
+
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException
+	{
+		// get the selected dumpFile
+		IResource dumpFile = EclipseHandler.getSelectedResource((IEvaluationContext) event.getApplicationContext());
+		IProject selectedProject = dumpFile.getProject();
+		IFile elfSymbolsFile = selectedProject.getFolder("build").getFile(selectedProject.getName().concat(".elf"));
+
+		activateIDFConsoleView();
+		List<String> commands = new ArrayList<String>();
+		commands.add(IDFUtil.getIDFPythonEnvPath());
+		commands.add(IDFUtil.getIDFSysviewTraceScriptFile().getAbsolutePath());
+		commands.add("-j");
+		commands.add("-b");
+		commands.add(elfSymbolsFile.getRawLocation().toOSString());
+		commands.add("file://".concat(dumpFile.getRawLocation().toString()));
+		console.println("Commands Prepared");
+		for (String command : commands)
+		{
+			console.println(command);
+		}
```

Comment:
For better readability can we display it in a single line?

Reply:
Done